templates/deploy-to-existing-k8s.yml

```diff
@@ -47,24 +40,24 @@ stages:
   jobs:
   - deployment: Deploy to Kubernetes
     displayName: Deploy job
-    environment: $(environmentName).$(k8sResourceName)
+    environment: '{{ environment.Environment.Name }}.{{ environment.Resources.kubernetesConnection.Name }}'
     strategy:
       runOnce:
         deploy:
           steps:
-          - task: Kubernetes@2
-            displayName: 'kubectl create imagepullsecret docker-auth'
+          - task: KubernetesManifest@0
+            displayName: kubectl create imagepullsecret docker-auth
             inputs:
-              containerRegistryType: 'Container Registry'
+              action: createSecret
+              secretName: docker-auth
               dockerRegistryEndpoint: $(dockerRegistryServiceConnection)
-              secretName: 'docker-auth'
               
           - task: KubernetesManifest@0
             displayName: Deploy to Kubernetes cluster
             inputs:
               action: deploy
-              namespace: $(k8sResourceNamespace)
+              namespace: $(k8sNamespace)
               manifests: |
                 manifests/*
               containers: |
-                $(containerRegistryName)/$(imageRepository)
+                $(containerRegistryName)/$(imageName)

```

Comment:
Change back to $(imageRepository). Introduce one more value at the top section called tag and set its value to $(Build.BuildId). 

While this is already the default in Docker task, still set it explicitly in Docker task so that it is clear to the users as most folks using templates may be new to Azure Pipelines YAML syntax. 

After that append  $(tag) to $(containerRegistryName)/$(imageRepository) in KubernetesManifest task